From 5c9c44274ecf0e704e1aadb1001d6df9f3ce403a Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Sun, 8 Feb 2026 21:25:06 -0800
Subject: [PATCH] dpkg-deb/build.c: Remove usage of --clamp-mtime in tar
 MIME-Version: 1.0 Content-Type: text/plain; charset=UTF-8
 Content-Transfer-Encoding: 8bit
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Recently dpkg added --clamp-mtime to tar to create reproducible
build tarballs [1].

But host tools doesn't support this option because is new on tar
so disable in our builds.

Signed-off-by: Aníbal Limón <anibal.limon@linux.intel.com>

Upstream-Status: Inappropriate [Configuration]

[1] https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=759999#20
[2] https://lists.gnu.org/archive/html/help-tar/2016-01/msg00000.html

Update patch context for dpkg 1.19.4.

Signed-off-by: Kai Kang <kai.kang@windriver.com>

Rebase to 1.23.5
Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 src/deb/build.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/src/deb/build.c b/src/deb/build.c
index 259c6251c..f7a0156f1 100644
--- a/src/deb/build.c
+++ b/src/deb/build.c
@@ -553,7 +553,6 @@ tarball_pack(const char *dir, filenames_feed_func *tar_filenames_feeder,
 		                 "-cf", "-",
 		                 "--format=gnu",
 		                 "--mtime", mtime,
-		                 "--clamp-mtime",
 		                 NULL);
 		/* Mode might become a positional argument, pass it
 		 * before -T. */
-- 
2.49.0

